census/estimator/trainer/task.py

```diff
@@ -0,0 +1,162 @@
+import argparse
+import json
+import os
+
+import model
+
+from tensorflow.contrib.learn.python.learn import learn_runner
+from tensorflow.contrib.learn.python.learn.utils import (
+    saved_model_export_utils)
+from tensorflow.contrib import metrics, learn
+
+
+def generate_experiment_fn(train_file,
+                           eval_file,
+                           num_epochs=None,
+                           train_batch_size=40,
+                           eval_batch_size=40,
+                           embedding_size=8,
+                           serving_input_fn_type='JSON',
+                           hidden_units=None,
+                           **experiment_args):
+  def _experiment_fn(output_dir):
+    train_input = model.generate_input_fn(
+        train_file, num_epochs=num_epochs, batch_size=train_batch_size)
+    eval_input = model.generate_input_fn(
+        eval_file,
+        batch_size=eval_batch_size,
+        skip_header_lines=1,
+        shuffle=False
+    )
+    if args.serving_input_fn_type == 'JSON':
```

Comment:
I don't think that's the right approach. The usecase for online prediction here is an app dev who is calling the serving backend using maybe data gleaned from an HTTP form, or a mobile client. Putting in the CSV format and shipping it off then parsing it again in the graph is a bunch of extra work for the user. JSON is better for app devs, and TFRecords are more efficient  for batch prediction... I add support for a binary that allows either of these approaches here: https://github.com/elibixby/cloudml-samples/commit/e89640d9977856dd031edbb94f52a34b0d58ef7a but there appears to be a bug in the service preventing me from merging.